core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +86,152 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
                 .collect(Collectors.toList());
     }
 
+    /** 
+     *  @return Data for market depth chart  
+    */
+    public MarketDepthInfo getMarketDepth(String currencyCode) {        
+        // Offer price can be null (if price feed unavailable), thus a null-tolerant comparator is used.
+        Comparator<Offer> offerPriceComparator = Comparator.comparing(Offer::getPrice, Comparator.nullsLast(Comparator.naturalOrder()));
+
+        // Trading btc-fiat is considered as buying/selling BTC, but trading btc-altcoin is
+        // considered as buying/selling Altcoin. Because of this, when viewing a btc-altcoin pair,
+        // the buy column is actually the sell column and vice versa. To maintain the expected
+        // ordering, we have to reverse the price comparator.
+        boolean isCrypto = CurrencyUtil.isCryptoCurrency(currencyCode);
+        if (isCrypto) offerPriceComparator = offerPriceComparator.reversed();
+
+        // Offer amounts are used for the secondary sort. They are sorted from high to low.
+        Comparator<Offer> offerAmountComparator = Comparator.comparing(Offer::getAmount).reversed();
+
+        var buyOfferSortComparator =
+                offerPriceComparator.reversed() // Buy offers, as opposed to sell offers, are primarily sorted from high price to low.
+                        .thenComparing(offerAmountComparator);
+        var sellOfferSortComparator =
+                offerPriceComparator
+                        .thenComparing(offerAmountComparator);
+
+        List<Offer> buyOffers = openOfferManager.getOpenOffersByCurrency(Direction.BUY.name(), currencyCode).stream().map(OpenOffer::getOffer).sorted(buyOfferSortComparator).collect(Collectors.toList()); 
+        List<Offer> sellOffers = openOfferManager.getOpenOffersByCurrency(Direction.SELL.name(), currencyCode).stream().map(OpenOffer::getOffer).sorted(buyOfferSortComparator).collect(Collectors.toList());
```

Comment:
good job Codacy, it spotted the copy-paste error here - buyOfferSortComparator vs sellOfferSortComparator

Reply:
fixed thanks